src/UpdateMetadataTrait.php

```diff
@@ -50,12 +50,17 @@ public function updateMetadata(
         $authUri = null,
         callable $httpHandler = null
     ) {
-        if (isset($metadata[self::AUTH_METADATA_KEY])) {
+        $metadata_copy = $metadata;
+        $metadata_copy = $this->applyMetricsHeader(
+            $metadata_copy,
+            $this->getServiceApiMetricsHeaderValue()
+        );
+
+        if (isset($metadata_copy[self::AUTH_METADATA_KEY])) {
             // Auth metadata has already been set
-            return $metadata;
+            return $metadata_copy;
```

Comment:
Discussed internally that we should indeed add header even if the auth token header is set because  invoking the `updateMetadata` method with auth token header set means that the intention is to explicitly just add metrics metadata into it.